#
# Patch for wscript: ensure package_revision_autogenerated.hpp is generated first.
# See: https://enkre.net/cgi-bin/code/qctool/info/5c7b688efb0a7209
#
diff -ru qctool-2.2.0.original/apps/wscript qctool-2.2.0.patched/apps/wscript
--- qctool-2.2.0.original/apps/wscript	2021-11-12 22:31:09.000000000 +0000
+++ qctool-2.2.0.patched/apps/wscript	2022-01-12 18:58:14.678676903 +0000
@@ -13,7 +13,7 @@
 
 external = [ "boost", "ZLIB", "PTHREAD", "CBLAS", "CLAPACK", "RT" ]
 components = [ 'SNPSummaryComponent', 'SampleSummaryComponent', 'HaplotypeFrequencyComponent', 'RelatednessComponent', 'SNPOutputComponent' ]
-base = [ 'appcontext', 'genfile', 'statfile', 'qcdb', 'metro', 'qctool-lib', 'package_revision' ]
+base = [ 'appcontext', 'genfile', 'statfile', 'qcdb', 'metro', 'qctool-lib', 'package_revision', 'config/package_revision_autogenerated.hpp' ]
 
 dependencies = {
 	"qctool": external + components + base,
diff -ru qctool-2.2.0.original/wscript qctool-2.2.0.patched/wscript
--- qctool-2.2.0.original/wscript	2021-11-12 22:31:09.000000000 +0000
+++ qctool-2.2.0.patched/wscript	2022-01-12 18:57:49.624609157 +0000
@@ -225,6 +225,7 @@
 		rule = compute_revision,
 	)
 
+	bld.add_group() # ensure the config above is created first
 	bld.stlib(
 		target = "qctool-lib",
 		source = bld.path.ant_glob( "src/*.cpp" ),
